calyx/opt/src/passes_experimental/fsm_builder.rs

```diff
@@ -0,0 +1,276 @@
+impl StaticSchedule<'_, '_> {
+    fn build_abstract(
+        &mut self,
+        scon: &ir::StaticControl,
+        guard: ir::Guard<ir::Nothing>,
+        mut transitions_to_curr: Vec<IncompleteTransition>,
+        looped_once_guard: Option<ir::Guard<ir::Nothing>>,
+    ) -> (Vec<IncompleteTransition>, Option<ir::Guard<ir::Nothing>>) {
+        match scon {
+            ir::StaticControl::Empty(_) => (transitions_to_curr, None),
+            ir::StaticControl::Enable(sen) => {
+                if matches!(sen.get_attributes().get(ACYCLIC), Some(1)) {
+                    // allocate one state per cycle
+                    // for all parts of the FSM that want to transition to this enable,
+                    // register their transitions in self.state2trans
+                    self.register_transitions(
+                        self.state,
+                        &mut transitions_to_curr,
+                        guard.clone(),
+                    );
+
+                    sen.group.borrow().assignments.iter().for_each(|sassign| {
+                        sassign
+                            .guard
+                            .compute_live_states(sen.group.borrow().latency)
+                            .into_iter()
+                            .for_each(|offset| {
+                                // convert the static assignment to a normal one
+                                let mut assign: ir::Assignment<ir::Nothing> =
+                                    ir::Assignment::from(sassign.clone());
+                                // "and" the assignment's guard with argument guard
+                                assign.and_guard(guard.clone());
+                                // add this assignment to the list of assignments
+                                // that are supposed to be valid at this state
+                                self.state2assigns
+                                    .entry(self.state + offset)
+                                    .and_modify(|other_assigns| {
+                                        other_assigns.push(assign.clone())
+                                    })
+                                    .or_insert(vec![assign]);
+                            })
+                    });
+                    self.state += sen.group.borrow().latency;
+                    // Don't know where to transition next; let the parent that called
+                    // `build_abstract_fsm_with_loop` deal with registering the transition
+                    // from the state(s) we just built.
+                    (
+                        vec![IncompleteTransition::new(
+                            self.state - 1,
+                            ir::Guard::True,
+                        )],
+                        looped_once_guard,
+                    )
+                } else {
+                    // must be cyclic otherwise
+                    // placeholder code for now.
```

Comment:
Is this supposed to be two sentences? As in: "Must be acyclic otherwise. The below is placeholder code for now."

As opposed to: "Must be acyclic; otherwise, run this placeholder code for now."